### PR TITLE
Improve performance for integer ingestion

### PR DESCRIPTION
**Pull request by Neverlord**
This PR implements lazy-padding for the range coder and switches to base 8 multi-level coding. Also fixes buggy `decode` results in the range encoder for operator `<` and `>=` (as revealed by the newly added unit tests).

### CODE REVIEW

File: libvast/vast/coder.hpp [hunk 1]
```diff
@@ -284,51 +343,43 @@ class range_coder : public vector_coder<Bitmap> {
...
     auto f = [=](auto& x) {
-      x.append_bits(true, n);
-      return true;
+      x.append_bits(true, (this->size_ + n) - x.size());
     };
-    auto result = std::all_of(this->bitmaps_.begin(), this->bitmaps_.end(), f);
-    if (result)
-      this->size_ += n;
-    return result;
+    std::for_each(this->bitmaps_.begin(), this->bitmaps_.end(), f);
```
[the remark this answers is not in the document]
Comment (Neverlord) [reply libvast/vast/coder.hpp]: The return type is more a question for @mavam. I suspect that `append` also doesn't actually need to return a boolean.

File: libvast/test/coder.cpp [hunk 2]
```diff
@@ -45,6 +48,17 @@ std::string dump(double x) {
...
 
+template <class Coder>
+void fill(Coder&) {
+  // end of recursion
+}
+
+template <class Coder, class T, class... Ts>
+void fill(Coder& c, T x, Ts... xs) {
+  c.encode(x);
+  fill(c, xs...);
```
[the remark this answers is not in the document]
Comment (Neverlord) [reply libvast/test/coder.cpp]: Damn right!

File: libvast/vast/coder.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (mavam) [reply libvast/vast/coder.hpp]: Yep, it's fine to use void here. OOM should be the only failure here, and a boolean is not the right way to capture this.

File: libvast/vast/coder.hpp [hunk 3]
```diff
@@ -168,24 +199,39 @@ class vector_coder : detail::equality_comparable<vector_coder<Bitmap>> {
...
   using typename super::value_type;
   using typename super::size_type;
+  using typename super::bitmap_type;
+
   using super::super;
 
+  bitmap_type& lazy_bitmap_at(size_t index) const {
+    auto& res = this->bitmaps_[index];
+    res.append_bits(false, this->size_ - res.size());
+    return res;
```
Comment (mavam) [inline libvast/vast/coder.hpp]: How about calling `res` here `result`? (Out style guide also mentions this.)

Comment (mavam) [reply libvast/vast/coder.hpp]: This comment applies to all occurrences of `res` in this PR.

File: libvast/vast/bitmap_index.hpp [hunk 4]
```diff
@@ -78,8 +78,8 @@ class bitmap_index
   /// the number of elements.
   /// @param n The number of elements to skip.
   /// @returns `true` on success.
```
[the remark this answers is not in the document]
Comment (Neverlord) [reply libvast/vast/bitmap_index.hpp]: Done.

**Pull request merged**